pkg/provider/azure_storageaccount.go

```diff
@@ -138,16 +141,40 @@ func (az *Cloud) GetStorageAccesskey(ctx context.Context, subsID, account, resou
 		return "", fmt.Errorf("empty keys")
 	}
 
+	var key string
+	var creationTime time.Time
+
 	for _, k := range *result.Keys {
 		if k.Value != nil && *k.Value != "" {
 			v := *k.Value
 			if ind := strings.LastIndex(v, " "); ind >= 0 {
 				v = v[(ind + 1):]
 			}
-			return v, nil
+			if !getLatestAccountKey {
+				// get first key
+				return v, nil
+			}
+			// get account key with latest CreationTime
+			if key == "" {
+				key = v
+				if k.CreationTime != nil {
+					creationTime = k.CreationTime.ToTime()
+				}
+				klog.V(2).Infof("got storage account key with creation time: %v", creationTime)
```

Comment:
nit: Do we need to print the creation time for every key? How about printing once in the end, before returning the key?

Reply:
there are actually only two keys: primary and secondary account keys, and the log could make sure if secondary key is rotated later, the secondary key should be retrieved, easier for troubleshooting. (by default, only one log is printed since the creation time is the same for both keys)